packages/experiments/src/components/Button/SplitMenuButton/SplitMenuButton.view.tsx

```diff
@@ -0,0 +1,64 @@
+/** @jsx withSlots */
+import { ContextualMenu, Stack, Text } from 'office-ui-fabric-react';
+import { withSlots, getSlots } from '../../../Foundation';
+import { Icon } from '../../../utilities/factoryComponents';
+
+import { Button } from '../Button';
+import { MenuButton } from '../MenuButton/MenuButton';
+import { ISplitMenuButtonComponent, ISplitMenuButtonProps, ISplitMenuButtonSlots } from './SplitMenuButton.types';
+
+export const SplitMenuButtonView: ISplitMenuButtonComponent['view'] = props => {
+  const {
+    styles,
+    tokens,
+    children,
+    primary,
+    disabled,
+    onClick,
+    expanded,
+    menu: Menu,
+    primaryActionDisabled,
+    onMenuDismiss,
+    menuTarget,
+    onSecondaryActionClick,
+    ...rest
+  } = props;
+
+  const Slots = getSlots<ISplitMenuButtonProps, ISplitMenuButtonSlots>(props, {
+    root: Stack,
+    button: Button,
+    menuButton: MenuButton,
+    stack: Stack,
+    icon: Icon,
+    content: Text,
+    menu: ContextualMenu,
+    menuIcon: Icon,
+    splitDivider: 'span'
+  });
+
+  return (
+    <Slots.root role="button" aria-disabled={disabled} horizontal as="span" verticalAlign="stretch">
+      <Slots.button
+        styles={styles}
+        tokens={tokens}
+        primary={primary}
+        disabled={primaryActionDisabled || disabled}
+        aria-disabled={primaryActionDisabled || disabled}
+        onClick={onClick}
+        {...rest}
+      >
+        {children}
+        <Slots.splitDivider />
```

Comment:
Hmm, interesting. Is there a rationale for having this inside of `Slots.button` instead of outside of it?

Reply:
I did it because in this way it was easy to style the border color of the divider with whatever the text color of the button was via tokens.